### ISSUE TITLE
keepassxc-cli export should direct password prompt to /dev/stderr

### ISSUE THREAD

**Issue by HUGHUSR** [keepassxreboot/keepassxc#4049]
Currently, `keepassxc-cli export` directs the password prompt to `/dev/stdout`. 
This reduces the ability to script an export by using a command such as:
`keepassxc-cli export -f xml -k ${keydir}/foo.key ${kdbxdir}/foo.kdbx > ${archive}/foo.xml`

Currently, no visible password prompt is displayed, and the password prompt is written as the first line of the XML file. If the password prompt was written to `/dev/stderr`, the password prompt would be displayed and the prompt would no longer be the fist line of the `foo.xml` file.

Writing the password prompt to `/dev/stderr` would also follow *nix conventions for prompts in scripted commands. (For example, like openssl, openssh, etc.)

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
Direct the password prompt for opening a database to `/dev/stderr`.
In fact, to follow *nix programming conventions, all prompts and errors should be directed to `/dev/stderr` and not to `/dev/stdout`.

## Current Behavior
[NOTE]: # ( Tell us what actually happens )
The password prompt is written to `/dev/stdout`, and therefore is written as the fist line of the xml file.

## Possible Solution
[NOTE]: # ( Not required, but suggest a fix/reason for the bug )

## Steps to Reproduce
[NOTE]: # ( Provide a link to a live example, or an unambiguous set of steps to )
[NOTE]: # ( reproduce this bug. Include code to reproduce, if relevant )
`keepassxc-cli export -f xml -k ${keydir}/foo.key ${kdbxdir}/foo.kdbx > ${archive}/foo.xml`

## Context
[NOTE]: # ( How has this issue affected you? What unique circumstances do you have? )
This makes it difficult to script archiving of KDBX files. Now, you have to write your own password prompt before executing the export command, and you have to edit the output file to strip off the password prompt. (Yeah, I could pipe the output to tail and strip off the first line that way... but it would make more sense to change to where the prompt is written.)

## Debug Info
[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.5.1
Revision: 0fd8836

Qt 5.13.1
Debugging mode is disabled.

Operating system: macOS Mojave (10.14)
CPU architecture: x86_64
Kernel: darwin 18.7.0

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- TouchID

Cryptographic libraries:
 libgcrypt 1.8.5

**Comment by droidmonkey** [comment]
I believe that is what `--quiet` is for. https://github.com/keepassxreboot/keepassxc/blob/ed0b76813d076376994bb030d4156eb769370187/share/docs/man/keepassxc-cli.1#L105

**Comment by HUGHUSR** [comment]
Granted, quiet would prevent the password prompt from being written to the file. But, it would still be preferable to have the command's password (and other prompts and errors) displayed (written to `/dev/stderr`) instead of the script having to create special code to handle such prompts.

As I previously noted, *nix convention is to write prompts to `/dev/stderr` for exactly that reason.

**Comment by droidmonkey** [comment]
That's exactly what quiet does.

**Comment by HUGHUSR** [comment]
I beg to differ with you.

I just tested with the `--quiet` option.

The `--quiet` option _**suppresses**_ the prompt. It does _**not**_ direct the prompt to `/dev/stderr`.

Neither
`keepassxc-cli export --quiet -f xml foo.kdbx > foo.xml`
nor
`keepassxc-cli export --quiet -f xml foo.kdbx > foo.xml 2> /dev/stderr`
show the password prompt. Both _**suppresses**_ the prompt. Neither show it. If `--quiet` were redirecting output to `/dev/stderr` I would see the prompt. Instead, it appears to be directing the output to `/dev/null`.

**Comment by droidmonkey** [comment]
Good point, will look into this further.

**Comment by louib** [comment]
@droidmonkey this is a duplicate of https://github.com/keepassxreboot/keepassxc/issues/3398

**Comment by HUGHUSR** [comment]
I agree that they are the same issue. 

I had searched on this topic before I created a new issue, and [#3398](https://github.com/keepassxreboot/keepassxc/issues/3398)  did not show up on the first several pages of the search and the results were becoming less relevant, so I assumed it was a new issue.

You will notice that we both use the same basic arguments as how prompts should be handled, too. Bottom line: Make sure all prompts and errors are written to `/dev/stderr` and not `/dev/stdout`.
